internal/providers/providers.go

```diff
@@ -287,11 +290,44 @@ func getCredentialForProvider(
 	}
 	if errors.Is(err, sql.ErrNoRows) {
 		zerolog.Ctx(ctx).Debug().Msg("no access token found for provider")
-		return credentials.NewEmptyCredential(), nil
+
+		// If we don't have an access token, check if we have an installation ID
+		return getInstallationTokenCredential(ctx, prov, store)
 	}
 	return nil, fmt.Errorf("error getting credential: %w", err)
 }
 
+// getInstallationTokenCredential returns a GitHub installation token credential if the provider has an installation ID
+func getInstallationTokenCredential(ctx context.Context, prov db.Provider, store db.Store) (provinfv1.Credential, error) {
+	installation, err := store.GetInstallationIDByProviderID(ctx, uuid.NullUUID{
+		UUID:  prov.ID,
+		Valid: true,
+	})
+	if err == nil {
+		cfg, err := githubapp.ParseV1Config(prov.Definition)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing github app config: %w", err)
+		}
+
+		privateKeyBytes, err := os.ReadFile(cfg.PrivateKeyPath)
+		if err != nil {
+			return nil, fmt.Errorf("error reading private key: %w", err)
+		}
+
+		privateKey, err := jwt.ParseRSAPrivateKeyFromPEM(privateKeyBytes)
```

Comment:
Are github credentials always RSA? is this something we can configure?

Reply:
They'll always be RSA. From the docs here: https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/managing-private-keys-for-github-apps#generating-private-keys
> Note: If you're using a library that requires a specific file format, the PEM file you download will be in PKCS#1 RSAPrivateKey format.